docs/source/considerations.rst

```diff
@@ -93,17 +93,38 @@ In fact, the only difference is the binding from Alice, in which we are using
 a lambda function for the handler.
 
 
+.. _shutting_down:
+
 Shutting down
 =============
 
-Although not covered in the examples until now (because many times you just
-want the multi-agent system to run forever until, perhaps, an event occurs),
-it is possible to actively kill the system using proxies:
+Sometimes, we will want to manually shut down a single agent or even a group
+of agents. In this section, we will see how to do it.
+
+Shutting down agents
+--------------------
+
+If we want to end the execution of a specific agent in our system, we can do
+it by calling the `agent.shutdown` method:
 
 .. literalinclude:: ../../examples/shutdown.py
 
-.. note:: Shutting down the nameserver will result in all agents registered in
-   the name server being shut down as well.
+Shutting down name servers
+--------------------------
+
+We learned earlier that name servers run just like any other agent in our
+system, even though with specific functionality.
+
+Shutting down a name server will result in all agents registered in that name
+server being shut down as well. This allows us to easily shutdown groups of
+agents at the same time, by registering them in different name servers when
+created.
+
+.. literalinclude:: ../../examples/various_nameservers.py
+
+.. note:: As seen in the example above, that two agents are registered on
+   different name servers does not mean that we cannot stablish connections
+   between them.
```

Comment:
Hmmmmm. I don't like this section very much...

I feel it is too complicated. Maybe the first paragraph ("Sometimes, we will want...") and the third paragraph ("We learned earlier...") could be completely removed, as they do not add useful information (in my opinion we should try to keep documentation clean and short). If we remove those paragraphs, then maybe we can remove the subsections too, and just keep a single "Shutting down" title/section.

The first example, which is shorter and simpler already shows the use of a single agent shutdown and the name server shutdown. Is it really necessary to add a more complex example just after? Couldn't we just note in the documentation how the `ns.shutdown()` call will shutdown all remaining registered agents in the name server? (just like before these changes)

I feel the second example is too complicated (timers, PUB-SUB, topics...) just to show how to use the shutdown methods. Maybe it is clear enough with the first example.

I wouldn't say no to adding a note (like the one you have there) in which we mention that "note how you can use multiple name servers to register different groups of agents in order to shutdown the whole group more easily. Of course, you do not need to have two agents registered in the same name server in order to create connections between them.". Or something like that. But I see that more like a note, just to make it clear, not something we should explicitly show with an example (I would assume the user understands how to do that if they want to).

What do you think?